### PR TITLE
bcm53xx: Re-enable Buffalo WZR-900DHP

### PR DESCRIPTION
**Pull request by shimamoto360**
Hi !

I found out about the WZR-900DHP being disabled by https://github.com/openwrt/openwrt/commit/e9672b1a8fa4714cbc35d5964caf704b2e571f35 via a Tweet by @musashino_205.
I have made a patch to re-enable it and will pull request it. https://github.com/openwrt/openwrt/pull/9567/commits/062bcd122ba6f9e95dbebbb8412d53300a9e5523

I squashed to get rid of unnecessary whitespace. https://github.com/openwrt/openwrt/commit/ae0aa73f31ed97a26a46c424f7dce3d75548637a

Re-enable the WZR-900DHP.
Specify the switch port in the DTS file.

Signed-off-by: SHIMAMOTO Takayoshi <takayoshi.shimamoto.360@gmail.com>

### CODE REVIEW

**Comment by shimamoto360** [comment]
@mans0n 
Thank you for your comments.

I had 22.03 as base because I wanted to raise the PR as an issue.
The reason was to ensure that the WZR-900DHP, which was supported in the 22.02 release, would continue to be supported.

I have not sent the PATCH to upstream yet.
The reasons are as follows.
WZR-1750DHP , WXR-1900DHP and other similar models have similar problems with DTS and have been added in the PATCH prepared by OpenWrt.
I have tried to match that approach.

Please point out if I am wrong.
Best regards.

**Comment by mans0n** [comment]
> WZR-1750DHP , WXR-1900DHP and other similar models have similar problems with DTS and have been added in the PATCH prepared by OpenWrt.
> I have tried to match that approach.

Thanks, I found it.
https://github.com/openwrt/openwrt/blob/103ad908c286049b0a7c68b793f2194b0c4d5711/target/linux/bcm53xx/patches-5.10/304-ARM-dts-BCM5301X-Specify-switch-ports-for-remaining-.patch
Then please move your changes to this patch file.

> The reason was to ensure that the WZR-900DHP, which was supported in the 22.02 release, would continue to be supported.

I can't say for sure, but if this can be merged to the master branch soon, then it might be backported to the 22.03 branch.
Anyway, this should go to the snapshot first, please rebase.

**Comment by shimamoto360** [comment]
@mans0n 
Thank you.

I have rebased.

Please point out if I have made a mistake.

Thank you for your continued support.

File: target/linux/bcm53xx/image/Makefile [hunk 1]
```diff
@@ -218,8 +218,7 @@ define Device/buffalo_wzr-900dhp
 	buffalo-enc WZR-900DHP2 $$(BUFFALO_TAG_VERSION) | \
 	buffalo-tag-dhp WZR-900DHP2 JP jp | buffalo-enc-tag | \
 	buffalo-dhp-image
-  BROKEN := y
-endef
+ endef
```
Comment (mans0n) [inline target/linux/bcm53xx/image/Makefile:221]: Remove the whitespace in this line.

**Comment by mans0n** [comment]
Please remove the 22.03 related commits. This PR should have only one commit by you.

And I'd prefer the commit message to be something like:
```
bcm53xx: add ports definition for WZR-900DHP

Buffalo WZR-900DHP was disabled in e9672b1a8fa4 ("bcm53xx: switch to the
upstream DSA-based b53 driver") due to its missing ports definition.
Specify the switch port in the DTS file and re-enable the device.

Signed-off-by: SHIMAMOTO Takayoshi <takayoshi.shimamoto.360@gmail.com>
```

**Comment by mans0n** [comment]
@rmilecki Could you please review this?

**Comment by shimamoto360** [comment]
@mans0n @rmilecki 
Thank you !!

**Pull request closed without merging**